#gaim grim's 64bit fixing patch v2
sha.h fixes yahoo on x86-64
everything else is for compile warnings

Index: src/gtkimhtml.c
===================================================================
RCS file: /cvsroot/gaim/gaim/src/gtkimhtml.c,v
retrieving revision 1.322
diff -u -p -r1.322 gtkimhtml.c
--- src/gtkimhtml.c	2 May 2004 18:12:07 -0000	1.322
+++ src/gtkimhtml.c	4 May 2004 07:26:28 -0000
@@ -553,7 +553,7 @@ static void gtk_imhtml_clipboard_get(Gtk
 	primary = gtk_widget_get_clipboard(GTK_WIDGET(imhtml), GDK_SELECTION_PRIMARY) == clipboard;
 
 	if (info == TARGET_HTML) {
-		int len;
+		gsize len;
 		char *selection;
 		GString *str = g_string_new(NULL);
 		if (primary) {
Index: src/sha.h
===================================================================
RCS file: /cvsroot/gaim/gaim/src/sha.h,v
retrieving revision 1.3
diff -u -p -r1.3 sha.h
--- src/sha.h	28 Sep 2003 01:10:06 -0000	1.3
+++ src/sha.h	4 May 2004 07:26:28 -0000
@@ -1,22 +1,41 @@
-#if (SIZEOF_INT == 4)
-typedef unsigned int uint32;
-#elif (SIZEOF_SHORT == 4)
-typedef unsigned short uint32;
-#else
-typedef unsigned int uint32;
-#endif /* HAVEUINT32 */
- 
-int strprintsha(char *dest, int *hashval);
- 
+/* 
+ * The contents of this file are subject to the Mozilla Public
+ * License Version 1.1 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of
+ * the License at http://www.mozilla.org/MPL/
+ * 
+ * Software distributed under the License is distributed on an "AS
+ * IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * rights and limitations under the License.
+ * 
+ * The Original Code is SHA 180-1 Reference Implementation (Compact version)
+ * 
+ * The Initial Developer of the Original Code is Paul Kocher of
+ * Cryptography Research.  Portions created by Paul Kocher are 
+ * Copyright (C) 1995-9 by Cryptography Research, Inc.  All
+ * Rights Reserved.
+ * 
+ * Contributor(s):
+ *
+ */
+#ifndef _SHA_H_
+#define _SHA_H_
+
+#include <glib.h>
+
 typedef struct {
-  unsigned long H[5];
-  unsigned long W[80];
-  int lenW;
-  unsigned long sizeHi,sizeLo;
+	guint32 H[5];
+ 	guint32 W[80];
+	int lenW;
+	guint32 sizeHi;
+	guint32 sizeLo;
 } SHA_CTX;
- 
+
+int strprintsha(char *dest, int *hashval);
 void shaInit(SHA_CTX *ctx);
 void shaUpdate(SHA_CTX *ctx, unsigned char *dataIn, int len);
 void shaFinal(SHA_CTX *ctx, unsigned char hashout[20]);
 void shaBlock(unsigned char *dataIn, int len, unsigned char hashout[20]);
-                                                                                                         
+
+#endif
Index: src/util.c
===================================================================
RCS file: /cvsroot/gaim/gaim/src/util.c,v
retrieving revision 1.283
diff -u -p -r1.283 util.c
--- src/util.c	21 Apr 2004 01:34:23 -0000	1.283
+++ src/util.c	4 May 2004 07:26:30 -0000
@@ -302,7 +302,8 @@ gaim_mime_decode_field(const char *str)
 
 	for (unencoded = cur = str; (encoded = cur = strstr(cur, "=?")); unencoded = cur) {
 		gboolean found_word = FALSE;
-		int i, num, len, dec_len;
+		int i, num, dec_len;
+		gsize len;
 		char *decoded, *converted;
 		char *tokens[3];
 
Index: src/protocols/oscar/oscar.c
===================================================================
RCS file: /cvsroot/gaim/gaim/src/protocols/oscar/oscar.c,v
retrieving revision 1.584
diff -u -p -r1.584 oscar.c
--- src/protocols/oscar/oscar.c	1 May 2004 22:17:43 -0000	1.584
+++ src/protocols/oscar/oscar.c	4 May 2004 07:26:35 -0000
@@ -5625,7 +5625,7 @@ static int oscar_send_chat(GaimConnectio
 	char *buf, *buf2;
 	char *charset = NULL;
 	int encoding;
-	int len;
+	gsize len;
 
 	if (!(conv = gaim_find_chat(gc, id)))
 		return -EINVAL;
Index: src/protocols/zephyr/zephyr.c
===================================================================
RCS file: /cvsroot/gaim/gaim/src/protocols/zephyr/zephyr.c,v
retrieving revision 1.90
diff -u -p -r1.90 zephyr.c
--- src/protocols/zephyr/zephyr.c	22 Apr 2004 01:53:18 -0000	1.90
+++ src/protocols/zephyr/zephyr.c	4 May 2004 07:26:36 -0000
@@ -578,7 +578,7 @@ static void handle_message(ZNotice_t not
 		GaimConversation *gconv1;
 		GaimConvChat *gcc;
 		char *ptr = notice.z_message + strlen(notice.z_message) + 1;
-		int len = notice.z_message_len - ((int)ptr - (int)notice.z_message);
+		int len = notice.z_message_len - (strlen(notice.z_message) +1);		
 		char *sendertmp = g_strdup_printf("%s", gaim_zephyr_get_sender());
 		GaimConvImFlags flags = 0;
 
